parsl/tests/test_bash_apps/test_stdout.py

```diff
@@ -87,8 +88,11 @@ def test_stdout_truncate(tmpd_cwd):
     assert len1 == 1
     assert len1 == len2
 
+    for record in caplog.records:
+        assert record.levelno < logging.ERROR
```

Comment:
These hunks are a good addition to these ATs; what about testing the exception paths as well?  Possibly a new test each for stderr and stdout.  Without thinking harder about it than ~30s, the first approach that comes to mind is a kernel like:
```python
assert "Could not parse stdout" in ...
# and
assert "Could not parse stderr" in ...
```
Or `mock.patch` the `get_std_fname_mode` and ensure that the mock was called.  Something like:
```python
exc_text = randomstring()  # See note below
mock_get_mode = mock.patch('...', side_effect=MemoryError(exc_text)  # an arbitrary exception
...
assert mock_get_mode.called
assert exc_text in ...
```

I don't think Parsl has implemented a `randomstring()` fixture, but [Compute has one we could borrow](https://github.com/globus/globus-compute/blob/2.18.0/compute_sdk/tests/unit/conftest.py#L11-L16).

Not mandatory, obviously, but sharing this here as a suggestion after talking out-of-band.